### PR TITLE
dotCMS/core#24456 pre-fill goal name 

### LINKED ISSUE #24456: [UI] Limit variant to 2 & remove description if empty  & pre-fill goal name
### Parent Issue

#24343

### User Story

- [x] Limit the MAX_VARIANTS_ALLOWED to 2
- [x] Remove description if empty
- [x] Pre-fill Goal name with `Primary Goal`
 
### Acceptance Criteria

- Experiments should support only one variant. 
- If the description is empty the container should be hidden and don't show an empty container.
- When creating a Goal, the name should be pre-filled with `Primary Goal`

### Proposed Objective

User Experience

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Make name dynamic based on goal selection if the user didn't set a custom name.

### CODE REVIEW

**Review by zJaaal — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts [hunk 1]
```diff
@@ -165,6 +172,30 @@ export class DotExperimentsConfigurationGoalSelectComponent implements OnInit, O
...
+
+    /**
+     * If the user don't set a custom name set the default name for the goal
+     * @param {string} typeValue
+     * @memberOf DotExperimentsConfigurationGoalSelectComponent
+     */
+    private defineDefaultName(typeValue: string): void {
+        const nameControl = this.form.get('primary.name') as FormControl;
+        if (
+            nameControl.value === this.DEFAULT_NAME_LABEL ||
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts:189]: If the user changes the input value manually, and clicks the options, do not change the value. It is ok in that way?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts [hunk 2]
```diff
@@ -77,6 +77,10 @@ export class DotExperimentsConfigurationGoalSelectComponent implements OnInit, O
         this.dotExperimentsConfigurationStore.goalsStepVm$;
     private destroy$: Subject<boolean> = new Subject<boolean>();
 
+    private BOUNCE_RATE_LABEL;
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts:80]: change it to 
```  
private BOUNCE_RATE_LABEL = this.dotMessageService.get(this.goals.BOUNCE_RATE.label);
private REACH_PAGE_LABEL = this.dotMessageService.get(this.goals.REACH_PAGE.label);
private DEFAULT_NAME_LABEL = this.dotMessageService.get(
        'experiments.configure.goals.name.default'
    );`

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts [hunk 3]
```diff
@@ -146,15 +150,18 @@ export class DotExperimentsConfigurationGoalSelectComponent implements OnInit, O
     }
 
     private initForm(): void {
+        this.BOUNCE_RATE_LABEL = this.dotMessageService.get(this.goals.BOUNCE_RATE.label);
```
Comment (oidacra) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts:153]: Delete those.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts (hunk 1 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts:189]: correct. that is the idea.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts (hunk 2 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts]: done

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts (hunk 3 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-goal-select/dot-experiments-configuration-goal-select.component.ts]: done

**Review by oidacra — APPROVED** [review]

**Pull request merged**